src/api/transaction.js

```diff
@@ -31,135 +36,41 @@ const Transaction = {
       senderAddress, // address
     } = args;
 
-    // args validation
     if (!type) {
-      throw new TypeError('type needs to be defined');
+      throw TypeError('type needs to be defined');
     }
     if (!senderAddress) {
-      throw new TypeError('senderAddress needs to be defined');
+      throw TypeError('senderAddress needs to be defined');
     }
-    if ((type === 'APPROVECREATEEVENT'
-      || type === 'BET'
-      || type === 'APPROVESETRESULT'
-      || type === 'APPROVEVOTE'
-      || type === 'TRANSFER')
+    if ((type === APPROVECREATEEVENT
+      || type === BET
+      || type === APPROVESETRESULT
+      || type === APPROVEVOTE
+      || type === TRANSFER)
       && (!token || !amount)) {
-      throw new TypeError('token and amount need to be defined');
+      throw TypeError('token and amount need to be defined');
     }
-    if ((type === 'APPROVESETRESULT' || type === 'APPROVEVOTE') && !topicAddress) {
-      throw new TypeError('topicAddress needs to be defined');
+    if ((type === APPROVESETRESULT || type === APPROVEVOTE) && !topicAddress) {
+      throw TypeError('topicAddress needs to be defined');
     }
-    if (type === 'APPROVEVOTE' && !oracleAddress) {
-      throw new TypeError('oracleAddress needs to be defined');
-    }
-
-    // Skip approve if enough allowance
-    let txType = type;
```

Comment:
why this deleted? since we don't need to do the `approve` and `create/setresult/vote` in two calls?

Reply:
Ya, we only show the Approve Tx for the tx cost now. There is a note for the user prompt that tells them they will have to confirm a follow-up tx.